src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirJsonInputFormatter.cs

```diff
@@ -95,8 +95,10 @@ public override Task<InputFormatterResult> ReadRequestBodyAsync(InputFormatterCo
 
                 if (delayedException != null)
                 {
+                    var errorMessage = string.IsNullOrEmpty(delayedException.Message) ? Api.Resources.ParsingError : delayedException.Message;
+
```

Comment:
@namadabu I haven't found a nice way to test this, since we aren't sure what the root cause of the null/empty exception messages is. Please let me know if you had something in mind.